### PR TITLE
[Encoding Extension] Added module outline

### PR DESCRIPTION
**Pull request by MovieStoreGuy**
**Description:**

Adding the required steps to help manage a new component and reduce the amount of noise generated in one PR.

**Link to tracking Issue:** 

https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/28686

**Testing:** 
NA

**Documentation:**

Added the rough outline of the component development to provide more of a public idea of the project outline. 
Not strictly tied to it since it is more a user facing version of the linked issue.

### CODE REVIEW

File: extension/encodingextension/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Encoding extension
+
```
Comment (atoulme) [inline extension/encodingextension/README.md:2]: Do you want to generate the status box?

Comment (MovieStoreGuy) [reply extension/encodingextension/README.md:2]: Yeah, sorry message on slack to add the details.

Comment (MovieStoreGuy) [reply extension/encodingextension/README.md:2]: Yeah, sorry message on slack to add the details.

**Review by atoulme — APPROVED** [review]

**Comment by atoulme** [comment]
LGTM, please rebase

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -86,7 +86,8 @@ exporter/zipkinexporter/                                 @open-telemetry/collect
 extension/asapauthextension/                             @open-telemetry/collector-contrib-approvers @jamesmoessis @MovieStoreGuy
 extension/awsproxy/                                      @open-telemetry/collector-contrib-approvers @Aneurysm9 @mxiamxia
 extension/basicauthextension/                            @open-telemetry/collector-contrib-approvers @jpkrohling @svrakitin
-extension/bearertokenauthextension/                      @open-telemetry/collector-contrib-approvers @jpkrohling @pavankrish123 @frzifus
+extension/bearertokenauthextension/                      @open-telemetry/collector-contrib-approvers @jpkrohling @pavankrish123
```
Comment (mx-psi) [inline .github/CODEOWNERS:89]: Removal of @frzifus looks unrelated to this PR. If it's necessary, let's do it in a separate PR?

Comment (MovieStoreGuy) [reply .github/CODEOWNERS:89]: That is my mistake trying to rebase

**Review by mx-psi — APPROVED** [review]
